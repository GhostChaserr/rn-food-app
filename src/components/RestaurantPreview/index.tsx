import React, { FC } from 'react'
import { ImageSourcePropType } from 'react-native'
import { icons } from 'src/constants'
import { RestaurantMenuT } from 'src/types'
import { APP_SCREENS } from '../Navigator/Navigator'

import {
  StyledTouchableView,
  StyledRestaurantImgWrapper,
  StyledRestaurantImg,
  StyledDeliveryTimeWrapper,
  StyledDeliveryTimeText,
  StyledRestaurantMetaWrapper,
  StyledRestaurantMetaHeader,
  StyledRestaurantMetaInfoWrapper,
  StyledRestaurantMetaInfoContainer,
  StyledRestaurantMetaInfoText,
  StyledStarIcon
} from './styles'

export type RestaurantProps ={
  data: {
    photo: ImageSourcePropType
    name: string
    rating: number 
    id: number
    menu: RestaurantMenuT[]
    duration: string
  }
  navigation?: any
}

const RestaurantPreview: FC<RestaurantProps> = (props) => {
  const handleRestaurantOpen = () => props.navigation.navigate(APP_SCREENS.RESTAURANT, { restaurantId: props.data.id })
  return (
    <StyledTouchableView onPress={handleRestaurantOpen}>
      <React.Fragment>
        <StyledRestaurantImgWrapper>
          <StyledRestaurantImg source={props.data.photo} />
          <StyledDeliveryTimeWrapper>
            <StyledDeliveryTimeText> {props.data.duration} min </StyledDeliveryTimeText>
          </StyledDeliveryTimeWrapper>
        </StyledRestaurantImgWrapper>
        <StyledRestaurantMetaWrapper>
          <StyledRestaurantMetaHeader>{props.data.name}</StyledRestaurantMetaHeader>
          <StyledRestaurantMetaInfoWrapper>
            <StyledRestaurantMetaInfoContainer>
              <StyledStarIcon source={icons.star} />
              <StyledRestaurantMetaInfoText>
                {props.data.rating}
              </StyledRestaurantMetaInfoText>
            </StyledRestaurantMetaInfoContainer>
            {props.data.menu.slice(0, 2).map(item => (
              <StyledRestaurantMetaInfoContainer key={item.menuId} style={{ marginLeft: 0 }}>
                <StyledRestaurantMetaInfoText>
                  {item.name}
                </StyledRestaurantMetaInfoText>
              </StyledRestaurantMetaInfoContainer>
            ))}
            <StyledRestaurantMetaInfoContainer>
              <StyledRestaurantMetaInfoText>$200</StyledRestaurantMetaInfoText>
            </StyledRestaurantMetaInfoContainer>
          </StyledRestaurantMetaInfoWrapper>
        </StyledRestaurantMetaWrapper>
      </React.Fragment>
    </StyledTouchableView>
  )
}

export default RestaurantPreview;
