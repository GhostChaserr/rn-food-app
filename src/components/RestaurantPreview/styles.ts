import styled from 'styled-components/native'


export const StyledTouchableView = styled.TouchableOpacity`
  margin: 5px 0;
`

export const StyledRestaurantImgWrapper = styled.View`
  margin-top: 5px; 
  position: relative;
  border-radius: 30px;
`

export const StyledRestaurantImg = styled.Image`
  width: 100%;
  height: 200px;
  border-radius: 30px;
`

export const StyledDeliveryTimeWrapper = styled.View`
  position: absolute;
  padding: 10px 40px;
  background-color: white;
  border-top-right-radius: 30px;
  border-bottom-left-radius: 30px;
  bottom: 0;
`

export const StyledDeliveryTimeText = styled.Text`
  font-size: 12px;
`

export const StyledRestaurantMetaWrapper = styled.View`
  margin-top: 10px;
`
export const StyledRestaurantMetaHeader = styled.Text`
  font-size: 24px;
`
export const StyledRestaurantMetaInfoWrapper = styled.View`
  margin-top: 5px;
  flex-direction: row;
`

export const StyledRestaurantMetaInfoContainer  = styled.View`
  margin-right: 5px;
  flex-direction: row;
`

export const StyledRestaurantMetaInfoText = styled.Text`
  font-size: 14px;
`

export const StyledStarIcon = styled.Image`
  width: 15px;
  height: 15px;
  margin-right: 5px;
  position: relative;
  top: 2px;
`