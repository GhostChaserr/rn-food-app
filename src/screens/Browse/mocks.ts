const categoriesMocks = [
  {
    id: 'plants',
    name: 'Plants',
    tags: ['products', 'inspirations'],
    count: 147,
    image: require('src/assets/icons/plants.png')
  },
  {
    id: 'seeds',
    name: 'Seeds',
    tags: ['products', 'shop'],
    count: 16,
    image: require('src/assets/icons/seeds.png')
  },
  {
    id: 'flowers',
    name: 'Flowers',
    tags: ['products', 'inspirations'],
    count: 68,
    image: require('src/assets/icons/flowers.png')
  },
  {
    id: 'sprayers',
    name: 'Sprayers',
    tags: ['products', 'shop'],
    count: 17,
    image: require('src/assets/icons/sprayers.png')
  },
  {
    id: 'pots',
    name: 'Pots',
    tags: ['products', 'shop'],
    count: 47,
    image: require('src/assets/icons/pots.png')
  },
  {
    id: 'fertilizers',
    name: 'fertilizers',
    tags: ['products', 'shop'],
    count: 47,
    image: require('src/assets/icons/fertilizers.png')
  }
]

export { categoriesMocks }
