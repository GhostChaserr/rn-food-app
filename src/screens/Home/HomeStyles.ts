import styled from 'styled-components/native'

export const StyledSafeAreaView = styled.SafeAreaView`
  flex: 1;
`
export const StyledContentWrapper = styled.View`
  flex: 1;
  padding: 20px;
`

export const StyledHeaderWrapper = styled.View`
  justify-content: space-between;
  flex-direction: row;
`

export const StyledTouchableIcon = styled.TouchableOpacity``

export const StyledHeaderIcon = styled.Image`
  width: 20px;
  height: 20px;
  position: relative;
  top: 5px;
`

export const StyledHeaderLocation = styled.View`
  justify-content: center;
  background-color: rgba(0,0,0, 0.1);
  padding: 10px 60px;
  border-radius: 10px;
`

export const StyledHeaderLocationText = styled.Text``

export const StyledHeadingsWrapper = styled.View`
  margin-top: 20px;
  flex-direction: column;
  align-items: flex-start;
`

export const StyledHeadingsHeader = styled.Text`
  font-weight: bold;
  font-size: 36px;
`

export const StyledCategoriesContainer= styled.View`
  margin-top: 10px;
`

export const StyledCategoryButton = styled.TouchableOpacity<{ isSelected: boolean }>`
  padding: 10px;
  padding-bottom: 20px;
  background-color: ${({ isSelected }) => isSelected ? '#FC6D3F;': 'white;' } 
  border-radius: 30px;
  align-items: center;
  justify-content: center;
  margin-right: 10px;
`

export const StyledCategoryWrapper = styled.View`
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
  background-color: white;
  border-radius: 30px;
`

export const StyledCategoryIcon = styled.Image`
  resize-mode: contain;
  width: 35px;
  height: 35px;
`

export const StyledCategoryTextWrapper = styled.View`
  margin-top: 10px;
  align-items: center;
  justify-content: center;
`

export const StyledFlatList = styled.FlatList`
 
`

export const StyledCategoryText = styled.Text<{ isSelected: boolean }>`
  font-size: 11px;
  color: ${({ isSelected }) => isSelected ? 'white;': 'black;' } 
  font-weight: bold;
`

export const StyledRestaurantsFeedWrapper = styled.View`
  margin-top: 10px;
`

