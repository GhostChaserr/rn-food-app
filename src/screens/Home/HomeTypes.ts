import { StackNavigationProp } from '@react-navigation/stack'
import { createStackNavigator } from '@react-navigation/stack'

type RootStackParamList = {
  Home: undefined;
  Profile: { userId: string };
  Feed: { sort: 'latest' | 'top' } | undefined;
};

const RootStack = createStackNavigator<RootStackParamList>();

type ProfileScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Profile'
>;


export type RestaurantCategory = {
  id: number
  icon: any
  name: string
}