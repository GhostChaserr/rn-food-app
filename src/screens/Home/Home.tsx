import React, { useState } from 'react'
import GlobalStyles from 'src/global.styles'
import { icons } from 'src/constants'

import { RestaurantPreviewT } from 'src/types'
import { RestaurantCategory, RestaurantPreview } from 'src/components'
import { RestaurantCategoryT } from 'src/components/RestaurantCategory'
import { categoryData, restaurantData } from './mocks'

import {
  StyledSafeAreaView,
  StyledHeaderWrapper,
  StyledTouchableIcon,
  StyledHeaderIcon,
  StyledHeaderLocation,
  StyledHeaderLocationText,
  StyledContentWrapper,
  StyledHeadingsWrapper,
  StyledHeadingsHeader,
  StyledCategoriesContainer,
  StyledFlatList,
  StyledRestaurantsFeedWrapper
} from './HomeStyles'

const Home = ({ navigation }: any) => {
  const [categories] = useState(categoryData)
  const [selectedCategory, setSelectedCategory] = React.useState<any>(null)

  const onCategorySelect = (category: RestaurantCategoryT) =>
    setSelectedCategory(category)

  const renderCategory = ({ item }: { item: any }) => {
    const restaurantCategory: RestaurantCategoryT = item
    return (
      <RestaurantCategory
        selected={selectedCategory?.id == restaurantCategory.id}
        icon={restaurantCategory.icon}
        onCategorySelect={onCategorySelect}
        id={restaurantCategory.id}
        name={restaurantCategory.name}
      />
    )
  }

  const renderRestaurantPreview = ({ item }: { item: any }) => {
    const restaurant: RestaurantPreviewT = item
    return (
      <RestaurantPreview
        navigation={navigation}
        data={{
          menu: restaurant.menu,
          duration: restaurant.duration,
          name: restaurant.name,
          id: restaurant.id,
          rating: restaurant.rating,
          photo: restaurant.photo
        }}
      />
    )
  }

  return (
    <StyledSafeAreaView style={GlobalStyles.adroidSafeArea}>
      <StyledContentWrapper>
        <StyledHeaderWrapper>
          <StyledTouchableIcon>
            <StyledHeaderIcon source={icons.nearby} />
          </StyledTouchableIcon>
          <StyledHeaderLocation>
            <StyledHeaderLocationText>
              {' '}
              45 Lincoln street{' '}
            </StyledHeaderLocationText>
          </StyledHeaderLocation>
          <StyledTouchableIcon>
            <StyledHeaderIcon source={icons.basket} />
          </StyledTouchableIcon>
        </StyledHeaderWrapper>

        <StyledHeadingsWrapper>
          <StyledHeadingsHeader>Main</StyledHeadingsHeader>
          <StyledHeadingsHeader>Categoris</StyledHeadingsHeader>
        </StyledHeadingsWrapper>

        <StyledCategoriesContainer>
          <StyledFlatList
            data={categories}
            horizontal
            showsHorizontalScrollIndicator={false}
            keyExtractor={(item: RestaurantCategoryT | any) => `${item.id}`}
            renderItem={renderCategory}
          />
        </StyledCategoriesContainer>
        <StyledRestaurantsFeedWrapper>
          <StyledFlatList
            data={restaurantData}
            keyExtractor={(item: RestaurantPreviewT | any) => `${item.id}`}
            renderItem={renderRestaurantPreview}
          />
        </StyledRestaurantsFeedWrapper>
      </StyledContentWrapper>
    </StyledSafeAreaView>
  )
}

export default Home
