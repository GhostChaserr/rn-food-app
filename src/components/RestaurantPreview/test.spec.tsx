import React from 'react'
import TestRenderer from 'react-test-renderer'
import { restaurantData } from 'src/screens/Home/mocks'
import RestaurantPreview from './index'

describe('Testing restaurant preview component', () => {
  it('Should render restaurant preview component', () => {
    const restaurant = restaurantData[0]
    const tree = TestRenderer.create(
      <RestaurantPreview
        navigation={{}}
        data={{
          menu: restaurant.menu,
          duration: restaurant.duration,
          name: restaurant.name,
          id: restaurant.id,
          rating: restaurant.rating,
          photo: restaurant.photo
        }}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
