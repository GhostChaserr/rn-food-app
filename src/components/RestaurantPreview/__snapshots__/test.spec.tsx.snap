// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing restaurant preview component Should render restaurant preview component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 5,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 30,
          "marginTop": 5,
          "position": "relative",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "borderRadius": 30,
            "height": 200,
            "width": "100%",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 30,
            "borderTopRightRadius": 30,
            "bottom": 0,
            "paddingBottom": 10,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 10,
            "position": "absolute",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 12,
            },
          ]
        }
      >
         
        30 - 45 min
         min 
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
          },
        ]
      }
    >
      ByProgrammers Burger
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 5,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginRight": 5,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 15,
                "marginRight": 5,
                "position": "relative",
                "top": 2,
                "width": 15,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          4.8
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginRight": 5,
            },
            Object {
              "marginLeft": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          Crispy Chicken Burger
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginRight": 5,
            },
            Object {
              "marginLeft": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          Crispy Chicken Burger with Honey Mustard
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          $200
        </Text>
      </View>
    </View>
  </View>
</View>
`;
